### PR TITLE
update nft marketplace schema

### PR DESCRIPTION
**Pull request by 0xbe1**

### CODE REVIEW

File: schema-nft-marketplace.graphql [hunk 1]
```diff
@@ -27,8 +27,8 @@ enum Network {
 
 enum NftStandard {
     ERC721
-    ERC721A
```
[the remark this answers is not in the document]
Comment (mrbrianhobo) [reply schema-nft-marketplace.graphql:30]: that makes sense, is there a way to tell if a contract is an erc721 variant easily (e.g. erc721a)? if not we can exclude for now and add in later.

File: schema-nft-marketplace.graphql [hunk 2]
```diff
@@ -162,10 +153,13 @@ type Trade @entity {
...
-    tokenId: BigInt!
+    " Token IDs of the traded NFT. "
+    tokenIds: [BigInt!]!
 
-    " Price (in ETH) "
+    " The amount of tokens to transfer for each tokenId. It is set at 1 except for ERC1155 batch. "
+    amounts: [BigInt!]!
+
+    " Price (in ETH). If only 1 tokenId is involved, then the price is determined by the token only. If the trade is incurred by a batch purchasing (available in x2y2), then the price is the average price in the batch. "
     priceETH: BigDecimal!
```
[the remark this answers is not in the document]
Comment (mrbrianhobo) [reply schema-nft-marketplace.graphql:163]: agreed, opensea has a similar concept with bundle sales (gem sweeps as well).

why do we need a list of amounts? how does a batch purchase work?

File: schema-nft-marketplace.graphql [hunk 3]
```diff
@@ -258,4 +255,4 @@ type CollectionDailySnapshot @entity {
 
     " Number of traded items of the day "
     dailyTradedItemCount: Int!
-}
+}
```
Comment (mrbrianhobo) [inline schema-nft-marketplace.graphql:258]: could you add a newline here?

File: schema-nft-marketplace.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xbe1) [reply schema-nft-marketplace.graphql:163]: @mrbrianhobo good to know! can you help me understand these questions about opensea:

- does opensea allow cross-collection bundle?
- does opensea allow in-collection though erc1155 bundle? let's say there is a erc1155 collection A, can I sell 2 A#1, and 3 A#2, in just one bundle?

File: schema-nft-marketplace.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xbe1) [reply schema-nft-marketplace.graphql:30]: @mrbrianhobo  i kinda feel if OpenSea is not differentiating erc721a from erc721, then we don't have to :)

this is a screenshot from an Azuki token. while Azuki is known to implement erc721a, OpenSea doesn't show that explicitly.

image

Check out https://opensea.io/assets/ethereum/0xed5af388653567af2f388e6224dc7c4b3241c544/2674

[the remark this answers is not in the document]
Comment (this-username-is-taken) [reply schema-nft-marketplace.graphql:30]: Agreed

File: schema-nft-marketplace.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (this-username-is-taken) [reply schema-nft-marketplace.graphql]: Why not make this scenario multiple `Trade`s?

[the remark this answers is not in the document]
Comment (0xbe1) [reply schema-nft-marketplace.graphql]: @this-username-is-taken how about this:

1. each Trade only corresponds to just 1 token ID
2. to handle batch purchasing, add `isBatchPurchase` to `Trade`; trade.id should be able to represent if a trade is in a batch, e.g., "{normal trade id}-{auto incrementing id within the batch (0, 1, 2, ...)}", another option is to have a `BatchTrade` entity.
3. for a batch purchase, divide the price evenly across all trades.

What do you think?

File: schema-nft-marketplace.graphql [hunk 4]
```diff
@@ -232,11 +232,11 @@ type CollectionDailySnapshot @entity {
     " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
     royaltyFee: BigDecimal!
 
-    " Number of tokens listed for sale in the marketplace. "
-    listedCount: BigInt!
+    " Minimum sale price of the day (in ETH) "
+    minSalePrice24hETH: BigDecimal!
```
Comment (this-username-is-taken) [inline schema-nft-marketplace.graphql:236]: why not `dailyMinSalePriceETH`? It'll be consistent with other schemas. Also 24h sounds like it's a rolling window

File: schema-nft-marketplace.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (this-username-is-taken) [reply schema-nft-marketplace.graphql]: What I'm trying to say is why not split this into multiple `Trade` entities? For example, in Bancor, we split a "trade" from USDC -> ETH as two Swaps, one USDC -> BNT, one BNT -> ETH. It's one tx but two events

[the remark this answers is not in the document]
Comment (0xbe1) [reply schema-nft-marketplace.graphql]: @this-username-is-taken exactly, i list some details if we were to split this into multiple `Trade` entities :)

**Comment by this-username-is-taken** [comment]
LGTM!

**Pull request merged**